packages/x-components/src/x-modules/facets-next/entities/__tests__/factory.spec.ts

```diff
@@ -0,0 +1,233 @@
+import { EditableNumberRangeFilter } from '@empathyco/x-types-next';
+import {
+  createNextEditableNumberRangeFilter,
+  createNextHierarchicalFilter,
+  createNextNumberRangeFilter,
+  createNextSimpleFilter,
+  createNumberRangeFilter,
+  createRawFilter,
+  createSimpleFilter
+} from '../../../../__stubs__/filters-stubs.factory';
+import { FiltersFactory } from '../factory';
+import { SingleSelectModifier } from '../single-select.modifier';
+import {
+  getStoreEditableNumberRangeFilter,
+  getStoreFilter,
+  getStoreFiltersByFacetId,
+  isEditableNumberRangeFilterSelected,
+  isFilterSelected,
+  prepareFacetsStore
+} from './utils';
+
+describe('testing filters entity factory', () => {
+  it('selects and deselects raw filters', () => {
+    const store = prepareFacetsStore();
+    const factory = new FiltersFactory();
+    const rawFilter = createRawFilter('size:m');
+    const rawFilterEntity = factory.createFilterEntity(store, rawFilter);
+
+    // Selecting a raw filter that is not in the store should add it.
+    rawFilterEntity.select(rawFilter);
+    expect(isFilterSelected(store, rawFilter.id)).toBe(true);
+
+    // Deselecting a raw filter should remove it from the store.
+    rawFilterEntity.deselect(rawFilter);
+    expect(store.state.x.facetsNext.filters).not.toHaveProperty(rawFilter.id);
```

Comment:
to make the same in all tests and not depends on the store structure:
```suggestion
    expect(getStoreFilter(store, rawFilter.id)).toBeUndefined();
```